lib/reverse_words.rb

```diff
@@ -1,4 +1,57 @@
 # A method to reverse each word in a sentence, in place.
+# def string_reverse(my_string)
+#
+#   if my_string == nil
+#     return nil
+#   elsif my_string.length == 1
+#     return my_string
+#   elsif
+#     my_string.length > 1
+#     temp = ""
+#     i = 0
+#     j = (my_string.length - 1)
+#     while i < j
+#       temp = my_string[i]
+#       my_string[i] = my_string[j]
+#       my_string[j] = temp
+#       i += 1
+#       j -= 1
+#     end
+#     return my_string
+#   end
+# end
+
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words == nil || my_words.length == 0
+    return nil
+  else
+    start = 0
+    ender = 0
+
+    my_words.length.times do |i|
+      ender = i
```

Comment:
To take care of multiple consecutive, white spaces, at the beginning of the times' loop you could check if my_words[start] is a white space. If it is, update, start to be i+1 and continue to the next iteration in the times' loop, otherwise, continue on with the rest of your logic.